src/cr/cube/matrix.py

```diff
@@ -80,24 +79,18 @@ class PrunedMatrix(_BaseTransformedMatrix):
 
     @lazyproperty
     def columns(self):
-        if not self._applied:
-            return self._base_slice.columns
-
         return tuple(
-            PrunedVector(column, self._base_slice.rows)
+            PrunedVector(column, self._base_slice.rows, self._applied)
             for column in self._base_slice.columns
-            if not column.pruned
+            if not column.hidden
         )
 
     @lazyproperty
     def rows(self):
-        if not self._applied:
-            return self._base_slice.rows
```

Comment:
I'm wondering why we can't also remove this `"prune==True"` (`self._applied`) test from `PrunedMatrix.table_base` and `PrunedMatrix.table_margin`? I'm thinking all knowledge of prune-or-not should live on the dimension and that we can remove `Transforms.prune`. Shouldn't those two produce the same result regardless of whether pruning is turned on? Or are we possibly still making some distinction between a pruned vector and a hidden one?